### DISCUSSION TITLE
is there a demo about asserting a bls12-381 signature is valid by gnark?

### DISCUSSION THREAD

**Discussion by xichan-ms** [Consensys/gnark#391] [Q&A]
as the title, thanks.

or ask for some paid help, I try to contact by mail zkteam@consensys.net, however, failed with wrong mail receiver, is there anyone can help? thanks.

**Comment by yelhousni** [comment]
We have BLS12-381 pairing here: https://github.com/ConsenSys/gnark/tree/develop/std/algebra/emulated/sw_bls12381
You can build on this to have a BLS signatyre circuit.
